apis/apps/v1alpha1/opsrequest_types.go

```diff
@@ -311,12 +311,33 @@ type Upgrade struct {
 	//
 	// This field is deprecated since v0.9 because ClusterVersion is deprecated.
 	//
-	// +kubebuilder:validation:Required
 	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.9.0"
-	ClusterVersionRef string `json:"clusterVersionRef"`
+	ClusterVersionRef string `json:"clusterVersionRef,omitempty"`
+
+	// Lists components to be upgrade based on desired ComponentDefinition or ServiceVersion.
+	// +patchMergeKey=componentName
+	// +patchStrategy=merge,retainKeys
+	// +listType=map
+	// +listMapKey=componentName
+	// +kubebuilder:validation:MaxItems=1024
+	Components []UpgradeComponent `json:"components,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"componentName"`
 }
 
-// VerticalScaling refers to the process of adjusting the compute resources (e.g., CPU, memory) allocated to a Component.
+// +kubebuilder:validation:XValidation:rule="has(self.componentDefinitionName) || has(self.serviceVersion)",message="at least one componentDefinitionName or serviceVersion"
+
+type UpgradeComponent struct {
+	// Specifies the name of the Component.
+	ComponentOps `json:",inline"`
+
+	// Specifies the name of the ComponentDefinition.
+	ComponentDefinitionName string `json:"componentDefinitionName,omitempty"`
+
+	// Specifies the version of the Service expected to be provisioned by this Component.
+	// Referring to the ServiceVersion defined by the ComponentDefinition and ComponentVersion.
```

Comment:
ServiceVersion in ClusterComponentSpec is optional, when no version is specified, use the latest available version in ComponentVersion.